### ISSUE TITLE
FXKeychain failed to retrieve data for key 'acess_token', error: -34018

### ISSUE THREAD

**Issue by kidsid49** [nicklockwood/FXKeychain#31]
This error comes sometimes while reading data from shared keychain by multiple apps. Can somebody explain what's causing this?

**Comment by altyus** [comment]
I'm having the same issue

**Comment by joekh** [comment]
me too FXKeychain failed to retrieve data for key

**Comment by 14lox** [comment]
+1

**Comment by 14lox** [comment]
Apple are aware of it... https://forums.developer.apple.com/thread/4743#14441

> UPDATE: We have finally been able to reproduce the -34018 error on iOS 8.3. This is the first step in identifying the root cause and then coming up with a fix.

> As usual, we can't commit to a release timeframe, but this has affected many developers and we really want to get this resolved.

> In the meantime, for a workaround, try adding a small delay in application:didFinishLaunchingWithOptions and applicationDidBecomeActive: before accessing the keychain. However, we know this hasn't worked in at least one case. Now that we have a reproducible case, I can experiment with the delay and see if I can come up with a recommended value.```

**Comment by Air-Craft** [comment]
Is it only when accessing the keychain right at launch? I'm getting this well after the app launch...

**Comment by emonster** [comment]
Having the same issue. Is there a solution to this problem? Thanks

**Comment by bmatthys** [comment]
Having the same issue and the workaround that Apple provided with the delay is not working for me either. Hope there will be a fix in a new iOS version soon though.

**Comment by maxoly** [comment]
Having the same issue only when running in debug mode with Xcode with the device connected to the debugger.

**Comment by sp3esu** [comment]
Having the same issue only when running in debug mode with Xcode/AppCode with the device connected to the debugger...

**Comment by piogab** [comment]
Same issue here in iOS 9.2.1. When device is not connected to the debugger it works ok for me.

**Comment by bimusiek** [comment]
Same issue with iOS 10 and iPhone 7 simulator...

**Comment by jboulter11** [comment]
@bimusiek and all others getting this issue:

If you're getting this issue now, it's likely because iOS 10 now requires a new entitlement for keychain usage.  Here's the fix:
https://forums.developer.apple.com/thread/60617

Go to your project target and under capabilities add Keychain Sharing to your app.  It immediately resolved my issues.

**Comment by bimusiek** [comment]
@jboulter11 Yes, you are right. I forgot to reply here after I found the solution. 

Confirmed, you have to enable Keychain Sharing, however for me this is not really the solution for people that do not want to share Keychain :) However, the fault is at Apple side.
